x-pack/plugins/secops/server/graphql/sources/source.mock.ts

```diff
@@ -5,13 +5,30 @@
  */
 
 /* tslint:disable */
+import { Logger } from '../../utils/logger';
+import { Context } from '../index';
 
-export default
-{
-  "id": "default",
-  "configuration": {
-    "fields": {
-      "host": "beat.hostname"
+export const mockSourceData = {
+  id: 'default',
+  configuration: {
+    fields: {
+      host: 'beat.hostname',
+    },
+  },
+};
+
+export const getSourceQueryMock = (logger: Logger | null) => ({
```

Comment:
Why would the logger be null ever? That would make it to where everywhere it's used to has to be checked for a null value? Wouldn't it always be injected for logging?

Reply:
At worst if it cannot inject a logger, you can use lodash's `noop` like so:

```ts
import noop from 'lodash/fp/noop'
export const getSourceQueryMock = (logger: Logger = noop) => . ({
```

And then you don't have to do any `logger &&` checks. Instead it will default to noops:
https://lodash.com/docs/4.17.10#noop

Make it easier reading

Reply:
Correction above, it actually is going to be probably a logger no-op object such as
```
Logger = { debug: noop, info: noop, warn: noop, error: noop }
```
used by default, but we hopefully can figure out why the unit test isn't injecting it today

Reply:
In Schema Test you can do this:

```ts
  const loggerMock = {
    debug: jest.fn(),
    warn: jest.fn(),
    error: jest.fn(),
    info: jest.fn(),
  };
  // Here we specify the return payloads of mocked types
  const mocks = {
    Query: () => ({
      ...getSourceQueryMock(loggerMock),
    }),
  };
```
And then you can make this strict again with regards to a non-nullable logger. If we can keep re-using the Logger mock or other injectables within the testing framework we can add a mocked logger next to the logger.

Reply:
ok get it